pkg/services/replicator/process.go

```diff
@@ -45,6 +49,11 @@ func (p *Replicator) HandleTask(ctx context.Context, task Task, res TaskResult)
 			stream = client.DemuxReplicatedObject(stream)
 		}
 	} else {
+		if err = p.checkExpiration(task.obj); err != nil {
```

Comment:
minor: this is not Replicator's concern imo, i'd make such exclusions on caller side

Reply:
objects are taken in a batch and handled via a pool of _some size_ in the Policer, so Replicator to me is the closest thing to the PUT/REPLICATE operation and is the best thing to abort operation if an object is expired b/w engine's SELECT and network replication. but i understand you, can move it to Policer @roman-khimov

Reply:
Caller doesn't have object header to judge on this IIUC. And replicator has to get it anyway.

Reply:
Ah, it can have it in this case. Then policer would be a better place to keep this code doing exactly one thing.

Reply:
@roman-khimov, @cthulhu-rider, i rechecked, a full obj can be here only in evacuation case, in a normal replication case it is now binary replication, so it is impossible to get expiration (or make additional unmarshaling after we make replication without unmarshaling 😋). so only evacuation can be improved (good, can be done but is almost useless to me), i would just drop this commit then (evacuation can be improved, but this is a control function, so even no logs can be done if smth is skipped)

Reply:
dropped commit